packages/bsky/src/data-plane/client/hosts.ts

```diff
@@ -0,0 +1,107 @@
+import { Etcd3 } from 'etcd3'
```

Comment:
Wish this turned-up better in the diff!  It is largely the contents of `packages/bsky/src/data-plane/client.ts` though there are some alterations so that it works with a `HostList` rather than a static list of host strings.

One change here to the retry logic is that we previously would attempt to retry on a separate host ip if any host on that ip failed.  I think this was a legacy behavior that we haven't needed for some time, and especially with etcd we'd like to defer to its host list.